kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -803,13 +803,13 @@ unifyEqualsNormalizedAc
                             allElements1
                             allElements2
                             Nothing
-                ([opaque], []) ->
+                ([opaque@(ElemVar_ _)], []) ->
                     lift $
                         unifyEqualsElementLists'
```

Comment:
Since this function always fails when `opaque` is not an `ElemVar`, please change the type of `unifyEqualsElementLists'`.